packages/app/src/background/services/connection/ConnectionService.ts

```diff
@@ -0,0 +1,170 @@
+    }
+
+    const connection = this.getIdentityConnection(commitment, identity.metadata);
+    this.connections.set(urlOrigin, connection);
+    await this.save();
+
+    await this.browserController.pushEvent(
+      { type: EventName.CONNECT_IDENTITY, payload: omit(connection, ["commitment"]) },
+      { urlOrigin },
+    );
+  };
+
+  disconnect = async (payload: unknown, { urlOrigin }: IZkMetadata): Promise<void> => {
+    await this.isOriginConnected(payload, { urlOrigin });
+    this.connections.delete(urlOrigin!);
+    await this.save();
+
+    await this.browserController.pushEvent(
+      { type: EventName.DISCONNECT_IDENTITY, payload: { urlOrigin } },
+      { urlOrigin },
+    );
+  };
+
+  isOriginConnected = (payload: unknown, { urlOrigin }: IZkMetadata): unknown => {
+    if (!urlOrigin) {
+      throw new Error("CryptKeeper: origin is not provided");
+    }
+
+    const isConnected = this.connections.has(urlOrigin);
+
+    if (!isConnected) {
+      throw new Error("CryptKeeper: origin is not connected");
+    }
+
+    return payload;
+  };
+
+  getConnections = (): Map<string, IIdenityConnection> => this.connections;
+
+  downloadStorage = (): Promise<string | null> => this.connectionsStorage.get<string>();
+
+  restoreStorage = async (data: BackupData | null): Promise<void> => {
+    if (data && typeof data !== "string") {
+      throw new Error("Incorrect restore format for connections");
+    }
+
+    await this.connectionsStorage.set(data);
+  };
+
+  downloadEncryptedStorage = async (backupPassword: string): Promise<string | null> => {
+    const data = await this.connectionsStorage.get<string>();
+
+    if (!data) {
+      return null;
+    }
+
+    const backup = this.cryptoService.decrypt(data, { mode: ECryptMode.MNEMONIC });
+    const encryptedBackup = this.cryptoService.encrypt(backup, { secret: backupPassword });
+
+    return this.cryptoService.generateEncryptedHmac(encryptedBackup, backupPassword);
+  };
+
+  uploadEncryptedStorage = async (backupEncryptedData: BackupData, backupPassword: string): Promise<void> => {
+    if (!backupEncryptedData) {
+      return;
+    }
+
+    const encryptedBackup = this.cryptoService.getAuthenticBackup(backupEncryptedData, backupPassword);
+
+    if (typeof encryptedBackup !== "string") {
+      throw new Error("Incorrect backup format for connections");
+    }
+
+    const backup = this.cryptoService.decrypt(encryptedBackup, { secret: backupPassword });
+    await this.unlock();
+
+    const backupAllowedHosts = new Map(JSON.parse(backup) as Iterable<[string, IIdenityConnection]>);
+    this.connections = new Map([...this.connections, ...backupAllowedHosts]);
+
+    await this.save();
+  };
+
+  private getIdentityConnection(commitment: string, metadata: IIdentityMetadata): IIdenityConnection {
+    return {
+      ...pick(metadata, ["name", "urlOrigin"]),
+      commitment,
+    };
+  }
+
+  private async save(): Promise<void> {
```

Comment:
```suggestion
  private async saveConnections(): Promise<void> {
```

Reply:
No need to specify it. Connection service works with connections.

Reply:
So maybe let's rename: 
```ts
private async saveApprovals(): Promise<void> {
    const serializedApprovals = JSON.stringify(Array.from(this.allowedHosts.entries()));
    const newApprovals = this.cryptoService.encrypt(serializedApprovals, { mode: ECryptMode.MNEMONIC });
    await this.approvals.set(newApprovals);
  }
```

`saveApprovals()` into `save()` inside the approval service. Its the same case.

Reply:
Sure, I can rename it in this PR
https://github.com/CryptKeeperZK/crypt-keeper-extension/pull/1090